feat: Changes to `project_id` or `role_id` will now result in the destruction and recreation of the `mongodbatlas_cloud_provider_access_authorization` resource (#3646)
mongodb/terraform-provider-mongodbatlas: docs/resources/cloud_provider_access.md

This is the second resource in the two-resource path as described above.
`mongodbatlas_cloud_provider_access_authorization`  Allows you to authorize an AWS or AZURE IAM roles in Atlas.

> -> **IMPORTANT:** Changes to `project_id` or `role_id` will result in the destruction and recreation of the authorization resource. This is because these fields uniquely identify the authorization and cannot be modified in-place.

Reviewer: ```suggestion
-> **IMPORTANT:** Changes to `project_id` or `role_id` will result in the destruction and recreation of the authorization resource. This action happens as these fields uniquely identify the authorization and cannot be modified in-place.
```

Reviewer 2: Addressed in [1c5ef3f](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/3646/commits/1c5ef3f41c2621b11380bb3b5b2a2d321a6be112)

## Argument Reference

* `project_id` - (Required) The unique ID for the project. **WARNING**: Changing the `project_id` will result in destruction of the existing authorization resource and the creation of a new authorization resource.
* `role_id`    - (Required) Unique ID of this role returned by mongodb atlas api. **WARNING**: Changing the `role_id` will result in destruction of the existing authorization resource and the creation of a new authorization resource.

> 

Reviewer: I discourage this approach, as it might not be as clear for the reader as the original one with the warning on each point.
